opampserver/pkg/connection/conncache.go

```diff
@@ -142,3 +143,12 @@ func (c *ConnectionsCache) UpdateAllConnectionConfigs(connConfigEvaluator func(c
 		}
 	}
 }
+
+func (c *ConnectionsCache) GetConnectionInfoByWorkload(podWorkload workload.PodWorkload) (*ConnectionInfo, error) {
+	for _, conn := range c.liveConnections {
```

Comment:
I think we should protect this `liveConnection` access with:

```go
	c.mux.Lock()
	defer c.mux.Unlock()
```

as it can be accessed by multiple goroutiens at the same time